lib/entice/logic/movement.ex

```diff
@@ -37,14 +37,28 @@ defmodule Entice.Logic.Movement do
     def init(%Entity{attributes: %{Movement => _}} = entity, _args),
     do: {:ok, entity}
 
-    def init(%Entity{attributes: %{Position => %Position{pos: pos, plane: plane}}} = entity, _args),
-    do: {:ok, entity |> put_attribute(%Movement{goal: pos, plane: plane})}
+    def init(%Entity{attributes: %{Position => %Position{coord: coord, plane: plane}}} = entity, _args),
+    do: {:ok, entity |> put_attribute(%Movement{goal: coord, plane: plane})}
 
     def init(entity, _args),
     do: {:ok, entity |> put_attribute(%Movement{})}
 
+    def handle_event({:movement_calculate_next},
+      %Entity{attributes: %{Movement => %Movement{update_self: update_self, update_delay: update_delay}}} = _entity) do
+      #TODO: implement once the whole collision business is handled
+      if update_self, do: start_update_trigger_timer(:movement_calculate_next, update_delay)
+    end
 
     def terminate(_reason, entity),
     do: {:ok, entity |> remove_attribute(Movement)}
+
+    defp start_update_trigger_timer(message, time) do
```

Comment:
I don't know. I copied it from here https://github.com/entice/logic/commit/20aa29d10dc6030774743f4a778c934d3ead72f9#diff-366987ce92dde211ad3e5f552c55e0d4R148 and assumed there was some kind of problem with send_after(message, 0).